pkg/vcs/linux.go

```diff
@@ -211,32 +218,63 @@ func (ctx *linux) addMaintainers(com *Commit) {
 	if len(com.CC) > 2 {
 		return
 	}
-	list := ctx.getMaintainers(com.Hash, false)
-	if len(list) < 3 {
-		list = ctx.getMaintainers(com.Hash, true)
+	mtrs := ctx.getMaintainers(com.Hash, false)
+	if len(mtrs) < 3 {
+		mtrs = ctx.getMaintainers(com.Hash, true)
 	}
-	com.CC = email.MergeEmailLists(com.CC, list)
+	com.Recipients = mtrs
 }
 
-func (ctx *linux) getMaintainers(hash string, blame bool) []string {
+func (ctx *linux) getMaintainers(hash string, blame bool) []RecipientInfo {
 	// See #1441 re --git-min-percent.
 	args := "git show " + hash + " | " +
 		filepath.FromSlash("scripts/get_maintainer.pl") +
-		" --no-n --no-rolestats --git-min-percent=20"
+		" --git-min-percent=20"
 	if blame {
 		args += " --git-blame"
 	}
 	output, err := osutil.RunCmd(time.Minute, ctx.git.dir, "bash", "-c", args)
 	if err != nil {
 		return nil
 	}
-	var list []string
-	for _, line := range strings.Split(string(output), "\n") {
-		addr, err := mail.ParseAddress(line)
+	lines := strings.Split(string(output), "\n")
```

Comment:
This is duplicated before both calls to this function, so I think it will be better for the function to accept whole output as []byte and split it onto lines by itself to eliminate the duplication (or maybe it won't need to split it in future, by e.g. using bufio.Scanner).
It will also make the test input easier to express with a raw string literal.